search-commons/src/test/java/no/unit/nva/search/resource/ResourceClientAllScientificValuesTest.java

```diff
@@ -0,0 +1,295 @@
+            .build()
+            .withFilter()
+            .apply()
+            .doSearch(resourceClient);
+    var hits = response.toPagedResponse().hits();
+    assertFalse(hits.isEmpty());
+  }
+
+  @Test
+  void shouldNotReturnDocumentsWhereScientificValuesAreMissingWhenMultipleProvidedValuesInRequest()
+      throws IOException, InterruptedException, BadRequestException {
+    var json =
+        """
+              {
+          "type": "Publication",
+          "identifier": "018ba3cfcb9c-94f77a1e-ac36-430a-84b0-0619e3bbaf39",
+          "entityDescription": {
+            "reference": {
+
+            }
+          }
+        }
+        """;
+    createIndexAndIndexDocument(json);
+
+    var response =
+        ResourceSearchQuery.builder()
+            .fromTestQueryParameters(
+                queryToMapEntries(ALL_SCIENTIFIC_VALUES_UNASSIGNED_AND_LEVEL_ZERO))
+            .withDockerHostUri(URI.create(container.getHttpHostAddress()))
+            .withRequiredParameters(FROM, SIZE)
+            .build()
+            .withFilter()
+            .apply()
+            .doSearch(resourceClient);
+    var hits = response.toPagedResponse().hits();
+    assertTrue(hits.isEmpty());
+  }
+
+  @Test
+  void shouldNotReturnDocumentsWhereOnlyOneScientificValueAndIsNotAsOneOfValuesAsProvidedInRequest()
+      throws IOException, InterruptedException, BadRequestException {
+    var json =
+        """
+                     {
+          "type": "Publication",
+          "identifier": "018ba3cfcb9c-94f77a1e-ac36-430a-84b0-0619e3bbaf39",
+          "entityDescription": {
+            "reference": {
+              "publicationContext": {
+                "publisher": {
+                  "type": "Publisher",
+                  "scientificValue": "LevelOne"
+                }
+              }
+            }
+          }
+        }
+        """;
+    createIndexAndIndexDocument(json);
+
+    var response =
+        ResourceSearchQuery.builder()
+            .fromTestQueryParameters(
+                queryToMapEntries(ALL_SCIENTIFIC_VALUES_UNASSIGNED_AND_LEVEL_ZERO))
+            .withDockerHostUri(URI.create(container.getHttpHostAddress()))
+            .withRequiredParameters(FROM, SIZE)
+            .build()
+            .withFilter()
+            .apply()
+            .doSearch(resourceClient);
+    var hits = response.toPagedResponse().hits();
+    assertTrue(hits.isEmpty());
+  }
+
+  private static void createIndexAndIndexDocument(String json)
+      throws IOException, InterruptedException {
+    indexingClient.createIndex(indexName, RESOURCE_MAPPINGS.asMap(), RESOURCE_SETTINGS.asMap());
+    indexingClient.refreshIndex(indexName);
+    var document =
+        new IndexDocument(
+            new EventConsumptionAttributes(indexName, SortableIdentifier.next()),
+            JsonUtils.dtoObjectMapper.readTree(json));
+    indexingClient.addDocumentToIndex(document);
+    Thread.sleep(DELAY_AFTER_INDEXING);
```

Comment:
This is probably not necessary and will be very slow. Can we do the same "force refresh" trick that NVI uses? Not sure, since the OpenSearch client is different there.

See `no.sikt.nva.nvi.index.aws.OpenSearchClientTest#refreshIndex` in `nva-nvi`.

Reply:
Forgot to remove this one, nice.